weasyprint/css/computed_values.py

```diff
@@ -176,6 +176,46 @@ def decorator(function):
     return decorator
 
 
+def compute_variable(value, name, computed, base_url, parent_style):
+    from .validation.properties import PROPERTIES
+    already_computed_value = False
+
+    if value and isinstance(value, tuple) and value[0] == 'var()':
+        variable_name, default = value[1]
+        computed_value = _resolve_var(computed, variable_name, default)
+        if computed_value is None:
+            new_value = None
+        else:
+            prop = PROPERTIES[name.replace('_', '-')]
+            if prop.wants_base_url:
+                new_value = prop(computed_value, base_url)
+            else:
+                new_value = prop(computed_value)
+
+        # See https://drafts.csswg.org/css-variables/#invalid-variables
+        if new_value is None:
+            try:
+                computed_value = ''.join(
+                    token.serialize() for token in computed_value)
+            except BaseException:
+                pass
+            LOGGER.warning(
+                'Unsupported computed value `%s` set in variable `%s` '
+                'for property `%s`.', computed_value,
+                variable_name.replace('_', '-'), name.replace('_', '-'))
+            if name in INHERITED and parent_style:
+                already_computed_value = True
+                value = parent_style[name]
+            else:
+                already_computed_value = name not in INITIAL_NOT_COMPUTED
+                value = INITIAL_VALUES[name]
+        elif isinstance(new_value, list):
+            value, = new_value
```

Comment:
The `content` property handler always returns a list regardless of input. In this case, the handler will only ever get a computed variable declaration, and thus I unpack the computed value again after getting it back from the handler.